apps/full-site-editing/full-site-editing-plugin/coming-soon/coming-soon.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Coming Soon
+ *
+ * @package A8C\FSE
+ */
+
+namespace A8C\FSE\Coming_soon;
+
+function show_coming_soon_page() {
+	global $post;
+
+	if ( is_user_logged_in() ) {
+		return false;
+	}
+
+	// Handle the case where we are not rendering a post.
+	if ( ! isset( $post ) ) {
+		return false;
+	}
+
+	// Allow anonymous previews
+	if ( isset( $_GET['preview'] ) ) {
+		return false;
+	}
+
+	return ( (int) get_option( 'page_on_front' ) === $post->ID );
```

Comment:
This gating can be removed, we can show the coming soon on all pages and not just on frontpage.